text/0031-integ-testing.md

```diff
@@ -0,0 +1,300 @@
+# Integration testing for CDK apps
+
+* **Original Author(s):**: [@nija-at](https://github.com/nija-at)
+* **Tracking Issue**: [#31](https://github.com/aws/aws-cdk-rfcs/issues/31)
+* **API Bar Raiser**: @{BAR_RAISER_USER}
+
+CDK app developers and construct library authors can write integration
+tests that can be run locally or in their CI/CD pipeline.
+
+## Working Backwards - README.md
+
+### contest
+
+🌩️ contest tests can verify that your CDK app can be deployed successfully on AWS with the desired resources.
+
+🤖 contest tests can verify that your CDK app functions as expected, such as assertions against a public API endpoint.
+
+🕰️ contest tests can verify that your CDK app can be updated from a previous version of the app.
+
+🧑‍💻 Write contest tests in any CDK supported language, and execute them as part of your CI pipeline.
+
+📸 Capture snapshots of your CDK apps to track how they are impacted by changes.
```

Comment:
Not described in the doc. I suppose we are leaving out the Custom Resource gunk from the snapshots?

Reply:
Yes, the doc is not complete. I plan to add this in subsequent revisions.